src/app-layer.c

```diff
@@ -620,6 +618,12 @@ static int TCPProtoDetect(ThreadVars *tv,
         }
     }
     SCReturnInt(0);
+parser_error:
+    ExceptionPolicyApply(p, g_applayerparser_error_policy, PKT_DROP_REASON_APPLAYER_ERROR);
+    SCReturnInt(-1);
+detect_error:
+    DisableAppLayer(tv, f, p);
+    SCReturnInt(-2);
```

Comment:
So if a parser enters an error state, a policy is applied, perhaps event to the flow.  On detect error, we disable the parser, will any policy be applied to those packets that will now no longer be parsed?

Reply:
policy for protocol detection exceptions isn't implemented. This change is just more clearly separate the errors by parser and pd

Reply:
so the above should change the existing pd exception behavior in any way